client/lxd_instances.go

```diff
@@ -442,13 +442,40 @@ func (r *ProtocolLXD) tryCreateInstance(req api.InstancesPost, urls []string, op
 	return &rop, nil
 }
 
+func (r *ProtocolLXD) isCurrentServer(server Server) (bool, error) {
+	if r == server {
+		return true, nil
+	}
+	if server == nil {
+		return false, fmt.Errorf("server is null")
+	}
+
+	serverInfo, err := server.GetConnectionInfo()
+	if err != nil {
+		return false, err
+	}
+
+	info, err := r.GetConnectionInfo()
+	if err != nil {
+		return false, err
+	}
+
+	return serverInfo.Protocol == info.Protocol &&
+		serverInfo.Certificate == info.Certificate &&
+		serverInfo.Project == info.Project, nil
+}
```

Comment:
I think it also needs to consider `clusterTarget` which currently isn't exposed in `ConnectionInfo`.

So I think we'd want to add a `Target string` to `ConnectionInfo`, expose it through `GetConnectionInfo` and then compare it in here.